plugin-server/functional_tests/plugins.test.ts

```diff
@@ -88,133 +91,119 @@ test.concurrent(`plugin method tests: event captured, processed, ingested`, asyn
-
-    const distinctId = new UUIDT().toString()
-    const uuid = new UUIDT().toString()
+        })
+        const teamId = await createTeam(organizationId)
+        const pluginConfig = await createAndReloadPluginConfig(teamId, plugin.id)
 
-    const event = {
-        event: 'custom event',
-        // NOTE: Before `sanitizeJsonbValue` was added, the null byte below would blow up the error
-        // UPDATE, breaking this test. It is now replaced with the Unicode replacement character,
-        // \uFFFD.
-        properties: { name: 'haha', other: '\u0000' },
-    }
+        const distinctId = new UUIDT().toString()
+        const uuid = new UUIDT().toString()
 
-    await capture({ teamId, distinctId, uuid, event: event.event, properties: event.properties })
+        const event = {
+            event: 'custom event',
+            properties: { name: 'haha', other: '\u0000' },
+        }
 
-    await waitForExpect(async () => {
-        const events = await fetchEvents(teamId)
-        expect(events.length).toBe(1)
-        return events
-    })
+        await capture({ teamId, distinctId, uuid, event: event.event, properties: event.properties })
 
-    const error = await waitForExpect(async () => {
-        const pluginConfigAgain = await getPluginConfig(teamId, pluginConfig.id)
-        expect(pluginConfigAgain.error).not.toBeNull()
-        return pluginConfigAgain.error
-    })
+        await waitForExpect(async () => {
+            const events = await fetchEvents(teamId)
+            expect(events.length).toBe(1)
+            return events
+        })
 
-    expect(error.message).toEqual('error thrown in plugin')
-    const errorProperties = error.event.properties
-    expect(errorProperties.name).toEqual('haha')
-    expect(errorProperties.other).toEqual('\uFFFD')
-})
+        const appMetric = await waitForExpect(async () => {
+            const errorMetrics = await fetchPluginAppMetrics(pluginConfig.id)
+            expect(errorMetrics.length).toEqual(1)
+            return errorMetrics[0]
+        })
 
-test.concurrent(`plugin method tests: creates error on unhandled rejection`, async () => {
-    const plugin = await createPlugin({
-        organization_id: organizationId,
-        name: 'test plugin',
-        plugin_type: 'source',
-        is_global: false,
-        source__index_ts: `
-            export async function processEvent(event) {
-                void new Promise((_, rejects) => { rejects(new Error('error thrown in plugin')) }).then(() => {})
-                return event
-            }
-        `,
-    })
-    const teamId = await createTeam(organizationId)
-    const pluginConfig = await createAndReloadPluginConfig(teamId, plugin.id)
+        expect(appMetric.successes).toEqual(0)
+        expect(appMetric.failures).toEqual(1)
+        expect(appMetric.error_type).toEqual('Error')
+        expect(JSON.parse(appMetric.error_details as string)).toMatchObject({
```

Comment:
I feel like there has to be a cleaner way to write this `JSON.parse(…)` to avoid getting a TypeScript error.

In Python, I'd do something similar to this to get both the explicit runtime check validation as well as type narrowing, rather than relying on `JSON.parse` to do something sensible on `undefined`:

```python
assert error_details is not None
assert json.loads(error_details) == ...
```

Is there some equivalent to that here?